client/src/components/WorkspaceDialog.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <v-dialog
+    v-model="dialog"
+    width="500"
+  >
+    <template v-slot:activator="{ on }">
+      <v-btn
+        class="workspace-btn ma-0 pa-4"
+        block
+        color="blue-grey darken-3"
+        dark
+        depressed
+        large
+        v-on="on"
+      >
+        New Workspace
+        <v-spacer />
+        <v-icon right dark>add_circle</v-icon>
+      </v-btn>
+    </template>
+
+    <v-card class="white--text">
+      <v-card-title
+        class="headline primary pt-3"
+        primary-title
+      >
+        Create Workspace
+      </v-card-title>
+
+      <v-card-text class="px-4 pt-4 pb-1">
+        <v-text-field
+          box
+          label="Workspace name"
+          v-model="newWorkspace"
+        />
+      </v-card-text>
+
+      <v-divider></v-divider>
+
+      <v-card-actions class="px-4 py-3">
+        <v-spacer></v-spacer>
+        <v-btn
+          color="blue-grey darken-3"
+          dark
+          depressed
+          @click="create"
+        >
+          Create Workspace
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script>
+import api from '@/api'
+
+export default {
+  data () {
+    return {
+      dialog: false
+    }
+  },
+  methods: {
+    async create () {
+      if (this.newWorkspace) {
+        const response = await api().post('multinet/graphql', {query: `mutation {
+          workspace (name: "${this.newWorkspace}" )
+        }`});
+
+        if (response.data.data) {
+          this.$router.push(`/workspaces/${this.newWorkspace}`);
+          this.dialog = false;
```

Comment:
There's an issue in the creation logic where the Sidebar component doesn't always get refreshed with the new workspace. I believe it *does* get updated when the view actually changes. That is, in the following situations, the sidebar will correctly update:
1. If the view is the Index/front page, and you create a workspace, the view switches to WorkspaceDetail for the new workspace, and the sidebar correctly refreshes.
2. If the view is WorkspaceDetail or GraphDetail or TableDetail, and you create a new workspace, *then switch to a different `*Detail` view*, then the sidebar will refresh.

I think the fix is to have the creation logic inform its parent that a new workspace has been created, but I am not totally sure of the best way to express this in Vue. I'd like to discuss it with @subdavis before we put in a fix.

Reply:
I spoke with @subdavis about this: he recommended one of three paths:
1. Use VueX and let the workspace list be updated from changes in application state.
2. Emit a "created" event from the new workspace dialog informing the sidebar of the newly created workspace name, and let it update its own internal state.
3. Emit a "please create" event from the new workspace dialog, and let the sidebar perform the creation and data update.

I think (1) is the most future-proof solution, since it can handle other kinds of application state as well, but since it doesn't seem to be needed right now I'm going with (2).

The problem with (3) is the question of what to do if the workspace creation fails for any reason; how and where do we then display the error condition? I would prefer that such errors simply appear in the modal dialog, so that if there's a fix, it can be offered there, or if not, the user can decide whether to try again or to abandon the action.